tests/installation/boot_into_snapshot.pm

```diff
@@ -23,38 +21,32 @@ use base "opensusebasetest";
 
 sub run {
     my ($self) = @_;
-    assert_screen 'linux-login', 200;
-    select_console 'root-console';
-    # 1)
-    assert_script_run('touch /etc/NOWRITE;test ! -f /etc/NOWRITE');
-    # 1b) just debugging infos
-    assert_script_run("snapper --iso list");
-    assert_script_run("cat /etc/os-release");
-    if (get_var("UPGRADE")) {
-        # if we made a migration, the version should be for example opensuse before migr. 42.1 > 42.2
-        # extract number of version id: example SlES 12.2 -> 12.2. for opensuse also ok
-        my $OS_VERSION     = script_output("grep VERSION_ID /etc/os-release | cut -c13- | head -c -2");
-        my $OLD_OS_VERSION = script_output("grep VERSION_ID /.snapshots/2/snapshot/etc/os-release | cut -c13- | head -c -2");
-        # grub_bug bug:956046. menu entry not stable. we could boot in wrong menu-entry.
-        if ($OS_VERSION eq $OLD_OS_VERSION) {
-            die "OS_VERSION after Rollback matches OS_VERSION before Rollback";
-        }
+    if (check_var('DESKTOP', 'gnome')) {
+        assert_screen('generic-desktop', 200);
+        select_console 'root-console';
+        script_run("systemctl reboot", 0);
+        # start from cdrom boot menu and it should boot from hard disk and select start bootloader for snapshot
+        assert_screen('bootloader', 200);
+        send_key 'ret';
+        # boot into leap snapshot
+        send_key_until_needlematch('start-bootloader-from-snapshot', 'down', 10, 5);
+        send_key 'ret';
+        send_key_until_needlematch('snapshot-before-upgrade', 'down', 30, 5);
+        send_key 'ret';
+        send_key_until_needlematch('opensuse-leap', 'down', 10, 5);
+        send_key 'ret';
+        save_screenshot;
+        reset_consoles;
+        assert_screen('generic-desktop', 200);
+    }
+    else {
+        assert_screen('linux-login', 200);
+        select_console 'root-console';
+        assert_script_run('touch /etc/NOWRITE;test ! -f /etc/NOWRITE');
+        script_run("systemctl reboot", 0);
+        reset_consoles;
+        $self->wait_boot;
```

Comment:
```suggestion
    my ($self) = @_;
    my $timeout = get_var('GRUB_TIMEOUT', 90);

    $self->handle_installer_medium_bootup;
    workaround_type_encrypted_passphrase;
    # 60 due to rare slowness e.g. multipath poo#11908
    # 90 as a workaround due to the qemu backend fallout
    assert_screen_with_soft_timeout('grub2', timeout => 2 * $timeout, soft_timeout => $timeout, bugref => 'boo#1120256');
    stop_grub_timeout;
    boot_into_snapshot;
    send_key_until_needlematch("bootmenu-xen-kernel", 'down', 10, 5) if get_var('XEN');
    if ((check_var('ARCH', 'aarch64') && is_sle && get_var('PLYMOUTH_DEBUG'))
        || get_var('GRUB_KERNEL_OPTION_APPEND'))
    {
        $self->bug_workaround_bsc1005313 unless get_var("BOOT_TO_SNAPSHOT");
    }
    else {
        # avoid timeout for booting to HDD
        send_key 'ret';
```

Reply:
http://10.162.23.47/tests/8377#step/boot_into_snapshot/3
I tested your suggestion in my PR, but this is not working.
what we have now is shutdown failed. 

As you said: alt-f2` is causing to switch to tty2 instead of opening desktop runner
So, gnome shorcuts may be broken after rollback That means that we would have caught a product bug

I need to debug this.

Reply:
there is no issue with alt-f2 during my debugging session, so no product issue so far.